Default Mission Config Breaks GOES & STIX TimeStream Recordings

#61 updated the default mission name from `swxsoc` to `demo`. This broke the timestream recording for mission-agnostic databases. The GOES and STIX data are loaded to the mission agnostic `swxsoc_sdc_aws_logs` however with the default mission name change, the executor tries to load them into `demo_sdc_aws_logs` which does not exist. 

```
|   ERROR: Failed to write to Timestream: An error occurred (ResourceNotFoundException) when calling the WriteRecords operation: The database demo_sdc_aws_logs does not exist. [swxsoc.db.timeseries]          |
| --- |
|   [ERROR] 2026-04-08T14:48:16.618Z Failed to write to Timestream: An error occurred (ResourceNotFoundException) when calling the WriteRecords operation: The database demo_sdc_aws_logs does not exist.       |
```

--- a/swxsoc/db/tests/test_timeseries.py
+++ b/swxsoc/db/tests/test_timeseries.py
@@ -16,18 +16,16 @@ from swxsoc.db import timeseries
 
 def get_test_db_names():
     """
-    Get mission-specific database and table names for testing.
+    Get database and table names for testing.
+
+    Delegates to :func:`~swxsoc.db.timeseries._get_timestream_names`.
 
     Returns
     -------
     tuple
-        (database_name, table_name) using current mission config
+        (database_name, table_name)
     """
-    import swxsoc
-
-    mission_name = swxsoc.config["mission"]["mission_name"]
-    database_name = f"dev-{mission_name}_sdc_aws_logs"
-    table_name = f"dev-{mission_name}_measures_table"
+    database_name, table_name, _ = timeseries._get_timestream_names()
     return database_name, table_name
 
 
@@ -543,3 +541,23 @@ def test_invalid_instrument_record_dimension_timestream(mocked_timestream):
     records = backend.databases[database_name].tables[table_name].records
 
     assert len(records) == 0
+
+
+@pytest.mark.parametrize("use_mission", ["demo"], indirect=True)
+def test_record_timeseries_uses_mission_agnostic_database_name(
+    use_mission, mocked_timestream
+):
+    """Regression test: default demo mission still writes to swxsoc_* resources."""
+    ts = TimeSeries(
+        time_start="2016-03-22T12:30:31",
+        time_delta=3 * u.s,
+        n_samples=1,
+        meta={"name": "agnostic_db_check"},
+    )
+    ts["temp4"] = [1.0] * u.deg_C
+
+    timeseries.record_timeseries(ts, instrument_name="test")
+
+    backend = timestreamwrite_backends[ACCOUNT_ID]["us-east-1"]
+    assert "dev-swxsoc_sdc_aws_logs" in backend.databases
+    assert "dev-demo_sdc_aws_logs" not in backend.databases

--- a/swxsoc/db/timeseries.py
+++ b/swxsoc/db/timeseries.py
@@ -21,6 +21,42 @@ import swxsoc
 __all__ = ["record_timeseries", "_record_dimension_timestream"]
 
 
+def _get_timestream_names() -> tuple[str, str, str]:
+    """
+    Return Timestream database, table, and mission names for the current environment.
+
+    The mission name is read from ``swxsoc.config["mission"]["mission_name"]``.
+    If the configured mission is ``demo`` (the library default), the name
+    ``swxsoc`` is used instead so that Timestream resources remain
+    mission-agnostic. For any other configured mission the real name is used
+    directly (e.g. ``hermes_sdc_aws_logs``).
+
+    Non-production environments are prefixed with ``dev-`` to match existing
+    deployment conventions.
+
+    Returns
+    -------
+    database_name : str
+        The Timestream database name, prefixed with ``dev-`` in non-production environments.
+    table_name : str
+        The Timestream table name, prefixed with ``dev-`` in non-production environments.
+    mission_name : str
+        The mission name used as a Timestream record dimension.
+    """
+    mission_name = swxsoc.config["mission"]["mission_name"]
+    if mission_name == "demo":
+        mission_name = "swxsoc"
+
+    database_name = f"{mission_name}_sdc_aws_logs"
+    table_name = f"{mission_name}_measures_table"
+
+    if os.getenv("LAMBDA_ENVIRONMENT") != "PRODUCTION":
+        database_name = f"dev-{database_name}"
+        table_name = f"dev-{table_name}"
+
+    return database_name, table_name, mission_name
+
+
 def record_timeseries(
     ts: TimeSeries, ts_name: str = None, instrument_name: str = ""
 ) -> None:
@@ -51,7 +87,11 @@ def record_timeseries(
     Notes
     -----
     Records are written in batches of 100 to comply with Timestream API limits.
-    Database and table names are automatically prefixed with 'dev-' when not in PRODUCTION environment.
+    Database and table names are derived from the configured mission name
+    (e.g. ``hermes_sdc_aws_logs``). When the default ``demo`` mission is
+    active, ``swxsoc`` is used instead so that Timestream resources remain
+    mission-agnostic. Names are automatically prefixed with 'dev-' when not
+    in PRODUCTION environment.
 
     NaN values are skipped entirely and not written to Timestream. When a NaN is encountered in the
     timeseries data, that specific measure value is omitted from the record. The function logs the
@@ -69,9 +109,6 @@ def record_timeseries(
     """
     timestream_client = boto3.client("timestream-write", region_name="us-east-1")
 
-    # Get mission name swxsoc config
-    mission_name = swxsoc.config["mission"]["mission_name"]
-
     # Validate Instrument name
     instrument_name = (
         instrument_name.lower()
@@ -87,12 +124,8 @@ def record_timeseries(
     if ts_name is None or ts_name == "":
         ts_name = ts.meta.get("name", "measurement_group")
 
-    # Get the Database and Table names based on Dev / Prod environment
-    database_name = f"{mission_name}_sdc_aws_logs"
-    table_name = f"{mission_name}_measures_table"
-    if os.getenv("LAMBDA_ENVIRONMENT") != "PRODUCTION":
-        database_name = f"dev-{database_name}"
-        table_name = f"dev-{table_name}"
+    # Get the Database, Table, and Mission names based on configured defaults and environment
+    database_name, table_name, mission_name = _get_timestream_names()
 
     dimensions = [
         {"Name": "mission", "Value": mission_name},
@@ -282,16 +315,8 @@ def _record_dimension_timestream(
         )
 
     try:
-        # Get mission name from environment or default to 'hermes'
-        mission_name = swxsoc.config["mission"]["mission_name"]
-
-        # Define database and table names based on mission and environment
-        database_name = f"{mission_name}_sdc_aws_logs"
-        table_name = f"{mission_name}_measures_table"
-
-        if os.getenv("LAMBDA_ENVIRONMENT") != "PRODUCTION":
-            database_name = f"dev-{database_name}"
-            table_name = f"dev-{table_name}"
+        # Resolve database and table names from mission-agnostic defaults.
+        database_name, table_name, _ = _get_timestream_names()
 
         record = {
             "Time": str(timestamp),

